components/style/gecko_conversions.rs

```diff
@@ -6,10 +6,50 @@
 //! Ideally, it would be in geckolib itself, but coherence
 //! forces us to keep the traits and implementations here
 
+#![allow(unsafe_code)]
+
 use app_units::Au;
+use gecko_bindings::bindings::{RawServoStyleSheetStrong, ServoComputedValuesStrong};
 use gecko_bindings::structs::nsStyleCoord_CalcValue;
+use gecko_bindings::sugar::refptr::{RefCounted, RawServoStyleSheetBorrowed, ServoComputedValuesBorrowed};
+use properties::ComputedValues;
+use std::mem::transmute;
+use std::ptr::null;
+use std::sync::Arc;
+use stylesheets::Stylesheet;
 use values::computed::{CalcLengthOrPercentage, LengthOrPercentage};
 
+macro_rules! unsafe_impl_refcounted {
+    ($strong:ident, $borrowed:ident, $servo:ident) => {
+        unsafe impl<'a> RefCounted<'a> for $servo {
+            type Strong = $strong;
+            type Borrowed = $borrowed<'a>;
+            fn into_strong(x: Arc<Self>) -> Self::Strong {
```

Comment:
These are all functions and not bare methods because of Rust's coherence rules. We'd be able to make them into functions (and often use `From` and `Into`) if gecko_bindings was just a module in the style crate.